### ISSUE TITLE
feature request: automatically build from sources some specific packages

### ISSUE THREAD

**Issue by taz-007** [actionless/pikaur#366]
Hi, 
In the past I used yaourt with customizepkg for some specific packages. This allowed to automatically rebuild from sources those specific packages whenever a new version was available (I'm talking about official packages, not those in aur).
Could a similar feature be implemented in pikaur ? Already being able to specify a list of packages to build from sources instead of using the official binary would be great. Specifying a list of patches to apply to the sources would be better but indeed that is more work.

Thanks!

**Comment by actionless** [comment]
it sounds like a `--devel` option, isn't it?

**Comment by taz-007** [comment]
I don't think so, unless I've misunderstood it. I'm not talking about rebuilding only the git/svn packages, but some regular packages (so not a git package). Even more so as I probably don't want to install the git version but prefer to use an official release.

**Comment by actionless** [comment]
ah, you mean to rebuild repo packages from source?

yeah, there is no such option, though you can do it by combining two pikaur operations:

`pikaur -G mpv && pikaur -P ./mpv/trunk/PKGBUILD`

**Comment by actionless** [comment]
pikaur already have `--rebuild` flag for AUR packages

i was thinking what it could be possible to implement handling that flag for repo packages and do the same as in example above with `pikaur -S --rebuild mpv`

**Comment by dhouck** [comment]
As for local changes to the package, if I understand right pikaur lets you make your own local changes to the Git repository containing the `PKGBUILD` file? That should let you add your own patches or change config options as long as Git can figure out how to merge those changes with the upstream.

(I might not understand right; I don't actually use pikaur yet. I'm looking for an AUR helper and pikaur is on my shortlist of ones to try)

**Comment by actionless** [comment]
@dhouck 
so far it can be done only manually in two steps (the ticket is about combining those two steps into one):

```sh
pikaur -G mpv && pikaur -P ./mpv/trunk/PKGBUILD
```

the `-G` command will clone the repo to the ./mpv directory inside `pwd`, however for repo packages under the hood it's using `asp` utility, which you also could use just directly

**Comment by dhouck** [comment]
I'm talking about the last sentence, "Specifying a list of patches to apply to the sources would be better but indeed that is more work.". That looks like it's already possible, even if the rest of it is multiple steps.

**Comment by actionless** [comment]
yeah, you can do it, inside PKGBUILD, next you can do a local commit, and next it will merge it locally during the pulling the newest changes if upstream package got updated

**Comment by RubenKelevra** [comment]
> ah, you mean to rebuild repo packages from source?
> 
> yeah, there is no such option, though you can do it by combining two pikaur operations:
> 
> `pikaur -G mpv && pikaur -P ./mpv/trunk/PKGBUILD`

This is rather complicated compared to a simple `yaourt -Sb $packagename`.

Also pikaur then thinks the package it's compiling is from the AUR:

`
$ pikaur -P ./go-ipfs/trunk/PKGBUILD 
Reading local package database...
Resolving AUR dependencies...
Reading repository package databases...

:: New dependency will be installed from repository:
 go-pie (for go-ipfs)                                       -> 2:1.13.7-1

:: AUR package will be installed:
 go-ipfs                               0.4.22-1             -> 0.4.22-1
`

I accidentally did this from the root user, which leads to this error: 

`
# pikaur -P ./go-ipfs/trunk/PKGBUILD 
Reading local package database...
:: error: failed to generate .SRCINFO from ./go-ipfs/trunk/PKGBUILD:
Running as unit: run-u23555.service
Finished with result: exit-code
Main processes terminated with: code=exited/status=200
Service runtime: 7ms

`
That's a bug I guess.
